### PR TITLE
Refactor: Add documentation for the Scastie configuration

### PR DESCRIPTION
**Pull request by Dedelweiss**
I have noticed that in most Scala projects using Scaladoc with snippets. When snippets were using external libraries, these did not work. Example:
Screenshot 2023-05-23 at 16 02 45

Planned or just lack of interest to look into it further, the question remains "How to fix it?".
So I noticed that a -scastie-configuration flag was present but as discussed in [Issue #16220](https://github.com/lampepfl/dotty/issues/16220) on Dotty, nothing explains how to use it or anything.
So when testing some things, I noticed that it was behaving like the "Add Extra Sbt Configuration" input of the Scastie web application. 

Screenshot 2023-05-23 at 16 06 32

And as soon as I added the following piece of code, the snippet worked as it should.

Screenshot 2023-05-23 at 16 10 04
Screenshot 2023-05-23 at 16 13 26

So the goal to this PR is to add some documentation about `-scastie-configuration` flag so the users can make the best use of Scastie.

Fixes: https://github.com/lampepfl/dotty/issues/16220

### CODE REVIEW

File: _overviews/scala3-scaladoc/settings.md [hunk 1]
```diff
@@ -172,6 +172,14 @@ Which means:
 - all snippets in files under directory `my/path/f` should fail during compilation
 - all other snippets should compile successfully
 
+#### -scastie-configuration
+
+Add the scastie-configuration flag to the Scala compiler options to enable the Scastie configuration. This will allow you to include additional Sbt configuration for your Scastie snippets. For example, when you import external libraries into your snippets, you need to add the related dependencies.
```
[the remark this answers is not in the document]
Comment (Dedelweiss) [reply _overviews/scala3-scaladoc/settings.md:177]: > Thank you for the investigation and the fix! Indeed, documenting this option seems important. There is something that is still unclear to me: how do we enable the “Scastie button”?

Hello, If I'm not mistaken and you are talking about the "play" button. It seems to me that it is activated when you put the `sc:compile` part. I can add it in the documentation but does it fit in the `flags` part or would a dedicated page be better?
Screenshot 2023-05-24 at 08 39 27
Screenshot 2023-05-24 at 08 39 35
Screenshot 2023-05-24 at 08 46 34

**Comment by Dedelweiss** [comment]
> Thank you for the investigation and the fix! Indeed, documenting this option seems important. There is something that is still unclear to me: how do we enable the “Scastie button”?

Hello, If I'm not mistaken and you are talking about the "play" button. It seems to me that it is activated when you put the `sc:compile` part. I can add it in the documentation but does it fit in the `flags` part or would a dedicated page be better?
Screenshot 2023-05-24 at 08 39 27
Screenshot 2023-05-24 at 08 39 35
Screenshot 2023-05-24 at 08 46 34

**Comment by Dedelweiss** [comment]
> Thank you for the improvement. I think at some point we should also document better the whole feature that adds this “play” button, but that could be done in a separate PR.

Thank you very much, I can do it if it's not a problem. I was thinking of either a page or a part [here](https://docs.scala-lang.org/scala3/guides/scaladoc/static-site.html) dedicated to the explanation of the possible parameters after the ```of a blockquote. Or else a page dedicated to the explanation of Scastie and its possible uses, I haven't found much of a page talking about it on docs.scala-lang (except briefly if I'm not mistaken).

**Comment by julienrf** [comment]
That would be great if you can do it! :pray: 

Is that feature related to the `-snippet-compiler:compile` option described [here](https://docs.scala-lang.org/scala3/guides/scaladoc/snippet-compiler.html)? If yes, I think the documentation could go to that page (maybe rename it to “Embedding code snippets” to make it more general, and make subsections like “snippets checking” and “interactive snippets”).

**Pull request merged**